File: 035-search-insert-position.go
//go:build ignore

package main

import "fmt"

func searchInsert(nums []int, target int) int {
	left := 0
	right := len(nums)
	for left < right {
		i := left + (right-left)/2
		if nums[i] == target {
			return i
		}
		if target < nums[i] {
			right = i
		} else {
			left = i + 1
		}
	}
	return left
}

func main() {
	data := []int{1, 3, 5, 6}
	fmt.Println(searchInsert(data, 6))
}

File: 035-search-insert-position.py
from typing import List


class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        l = 0
        r = len(nums) - 1
        while l <= r:
            i = l + (r - l) // 2
            if nums[i] == target:
                return i
            if nums[i] < target:
                l = i + 1
            else:
                r = i - 1
        return l


print(Solution().searchInsert([1, 3, 5, 7], 9))
